pkg/provider/provider.go

```diff
@@ -159,7 +166,9 @@ func (p *Provider) buildServicesAndRoutersForService(t *topology.Topology, cfg *
 		for _, svcPort := range svc.Ports {
 			entrypoint, err := p.buildTCPEntrypoint(svc, svcPort.Port)
 			if err != nil {
-				p.logger.Errorf("Unable to build TCP entrypoint for Service %q and port %d: %v", svcKey, svcPort.Port, err)
+				svc.Err = fmt.Errorf("unable to build TCP entrypoint for port %d: %v", svcPort.Port, err)
```

Comment:
Here we loop over ports in the service, and if there is an error, we set the error on the service...but what happens of there are 2 ports that cause errors? Do we care if there are more than one error? Should `svc.Err` be a slice that the error gets appended to?

Reply:
Indeed, actually we are about to change the type of the ```Err``` from ```error``` to ```[]string``` because we realize that it can be an issue if we want to marsh the topology.
It will solve these 2 issues.